lnwire/update_add_htlc.go

```diff
@@ -68,29 +73,39 @@ var _ Message = (*UpdateAddHTLC)(nil)
 //
 // This is part of the lnwire.Message interface.
 func (c *UpdateAddHTLC) Decode(r io.Reader, pver uint32) error {
-	return ReadElements(r,
+	err := ReadElements(r,
 		&c.ChanID,
 		&c.ID,
 		&c.Amount,
 		c.PaymentHash[:],
 		&c.Expiry,
 		c.OnionBlob[:],
 	)
+	if err != nil {
+		return err
+	}
+
+	return c.ExtraData.Decode(r)
```

Comment:
can we add `ExtraOpaqueData` to the codec? then the diff is only two lines for most cases